src/rollup/index.ts

```diff
@@ -345,7 +362,149 @@ export default function rollup (rawInputOptions: GenericConfigObject) {
 
 				return result;
 			});
+
+		return graph.buildChunks(inputOptions.input)
```

Comment:
My hope is that if we can manage to make code-splitting always on (as outlined above), we could basically remove the first version of `generate`.

Reply:
Perhaps a middle ground for always-on could be that one input and an output file would output chunks in the same folder as the output file, possibly separating the dynamic import inlining option as `inlineDynamicImports: true` or similar?

Reply:
Ah I see, I was forgetting about this. I guess a flag would definitely make sense as otherwise the behaviour would be inconsistent.

Reply:
So maybe:
* If we have a single entry point
  * if `output.file` is provided, it is the name of the chunk generated for the entry point
  * if a dynamic import is encountered + `inlineDynamicImports: true`, it is of course inlined
  * if a dynamic import is encountered + `inlineDynamicImports: false`
    * if `output.dir` is provided, additional generated chunks are placed there
    * otherwise, an error is shown
* If we have multiple entry points
  * if `output.dir` is missing, we show an error
  * if `output.file` is provided, we show a warning that it is ignored
?

Reply:
I noticed the `experimentalCodeSplitting` flag isn't really behaving like an "experimental" flag here since it's guiding the whole process. I've made the following change for now:

* If `input` is a string, do a single file build, assuming `file` and warning otherwise
* If `input` is an array and the flag is enabled, do a code splitting build, assuming `dir` and warning otherwise

We can potentially look at different arrangements of this before flag removal, but at least the flag itself is not part of the final decision now.